### ISSUE TITLE
Unfree license

### ISSUE THREAD

**Issue by puppe** [UltimateHackingKeyboard/agent#1276]
Hi @mondalaci,

first of all, the Ultimate Hacking Keyboard looks amazing. I am seriously considering buying one, as I am looking for a customizable, ergonomic keyboard. The UHK really seems to check almost all the boxes. Thank you for your fantastic work on the hardware and the software!

I couldn't help but notice though that you changed the license for Agent and the firmware  from GPL 3 to a short, custom license back in November of 2019. As far as I can tell this custom license is an unfree license. The first condition prohibits anyone from distributing a modified version that can be used with anything but the Ultimate Hacking Keyboard. So, while the software is still open-source it is not free/libre anymore.

You are of course free to license your work as you seem fit, and you don't owe any explanation. Nevertheless, as a prospective customer, I would be very interested in the reasons for this change. For me, the new license does make the UHK less attractive. It would also certainly dissuade me from contributing to the software.

Best regards,
Martin

**Comment by mondalaci** [comment]
Hi @puppe,

The change of license is the consequence of the behavior of our direct competitors who copied the UHK design very closely, and in some ways rather blatantly. We've been pouring considerably more resources into our configurator and firmware than they did regarding their products, and we don't want them to leverage our work.

I'm using GNU/Linux all day long exclusively, so I do understand how a non-free license must make you feel, and I'm sorry if this turns out to be a deal breaker for you, however, all things considered, the current license makes the most sense to us in this business environment.

Thank you for your understanding. I'm closing this issue, but feel free to follow up regardless.

**Comment by benneti** [comment]
Just wanted to say, that using the free license (GPL) was the reason that I bought two UHKs in the first place.

**Comment by leo-lb** [comment]
@mondalaci 

Let me suggest an alternative strategy because this is really really unfortunate. For both software and firmware.

You can withdraw public access to the source code and only provide source code to individual customers under the GPL license. Then you can introduce a sales contract where you specify various terms under which you reserve the right to stop distributing future updates to the source code under GPL to that particular customer. This way you can find out about customers who are competitors and bought your keyboard solely to get access to the source code for their product and you can selectively stop providing them with updates to the source code.

Please adopt this strategy or another to keep the code under a Free Software (ideally GPLv3+) license.
Thank you

**Comment by leo-lb** [comment]
You also can't say it is an "open source" keyboard with the current license, it is not true.

See: https://opensource.org/osd

```
6. No Discrimination Against Fields of Endeavor
The license must not restrict anyone from making use of the program in a specific field of endeavor. For example, it may not restrict the program from being used in a business, or from being used for genetic research.
```

**Comment by mondalaci** [comment]
When people refer to open source, they either mean it in a loose sense or a strict sense. In a loose sense, it means that the source is available in the open. In a strict sense, it means that it's licensed according to an OSI-approved license. We mean the former.

I don't want to withdraw public access to the source code because I consider it a practical advantage.

**Comment by kareltucek** [comment]
@leo-lb If I may chime in with my (possibly strong) opinion, I think I don't understand, since it seems to me that your proposal does not make current situation any more liberal, but on the contrary makes it much more restrictive.

In my vocabularly, words "free", "closed" and "open" are part of plain english and are defined by the way people use them and understand them. I.e., no arbitrary group of people can claim a right to define their meaning. Furthermore, in my vocabulary:

- "open" (source) means publicly available (without having to ask anyone) albeit with some restrictions on rights to use, modify or distribute
- "free" means being free to use, modify and distribute without any further restrictions

In this respect situation after your proposal is neither "free" nor "open" [^1]. On the other hand current situation is that the code is "open", and within the use inside of the domain of UHK community even "free" to some extent.

Also I do not understand how would I reditribute the source code and firmware executables under your model. If I understand you correctly - everyone wanting to try out my firmware would have to personally write me an email. I would have to forward that email to Laci in order to request formal permission to provide the requested executable, and then to mail the firmware file to the person who asked for it. Furthermore, I would have to distribute in this way not just the firmware itself, but as well all its update. This is madness.

TL;DR; I am fine with current situation, because it does not restrict me in any way from modifying and redistributing neither source code nor binaries. I agree that it is not ideal (since the ideal state is being both free and open source, with projects sharing knowledge worldwide for greater good), but I do understand the motivation for making it unfree". Also, I strongly prefer short and vague licence which is easy to understand over non-understable heaps of laywer talk.

[^1]: Not open because something that is not even publicly available is definitely not open. With respect to "free", apart from not being open, I believe that GPL is not a free license - correct me if GPLv3+ is somehow different from GPLv<2 in this respect.

**Comment by benneti** [comment]
I think it could be a good compromise to add a clause that in some years the source code can be used under the GPL.
This would ensure that no one can steal the current tech (I completely understand this fear) but would ensure that users that value free software more know that in a few years they can use the UHK completely under a FSF approved license.

**Comment by leo-lb** [comment]
@mondalaci Microsoft uses "Source available" or "Shared source" for some of it's code published under nonfree licenses.
